devcontainers/spec/devcontainers/devcontainers/file_parser_spec.rb

```diff
@@ -0,0 +1,208 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+require "dependabot/dependency_file"
+require "dependabot/source"
+require "dependabot/devcontainers/file_parser"
+require "dependabot/devcontainers/requirement"
+require_common_spec "file_parsers/shared_examples_for_file_parsers"
+
+RSpec.describe Dependabot::Devcontainers::FileParser do
+  it_behaves_like "a dependency file parser"
+
+  let(:parser) do
+    described_class.new(dependency_files: files, source: source, repo_contents_path: repo_contents_path)
+  end
+
+  let(:source) do
+    Dependabot::Source.new(
+      provider: "github",
+      repo: "mona/Example",
+      directory: directory
+    )
+  end
+
+  let(:files) do
+    [
+      config_file
+    ]
+  end
+
+  let(:repo_contents_path) { build_tmp_repo(project_name, path: "projects") }
+
+  let(:config_file) do
+    Dependabot::DependencyFile.new(
+      name: config_file_name,
+      content: fixture("projects", project_name, directory, config_file_name)
+    )
+  end
+
+  let(:dependencies) { parser.parse }
+
+  shared_examples_for "parse" do
+    it "parses dependencies fine" do
+      expectations.each.with_index do |expected, _index|
+        version = expected[:version]
+        name = expected[:name]
+
+        expect(dependencies.size).to eq(expectations.size)
+
+        dependency = dependencies.find { |dep| dep.name == name }
+        expect(dependency.name).to eq(name)
+        expect(dependency.version).to eq(version)
+        expect(dependency.requirements).to eq(expected[:requirements])
+        expect(dependency.metadata).to eq(expected[:metadata])
+      end
+    end
+  end
+
+  context "with a .devcontainer.json in repo root" do
+    let(:project_name) { "config_in_root" }
+    let(:config_file_name) { ".devcontainer.json" }
+    let(:directory) { "/" }
+
+    let(:expectations) do
+      EXPECTED_PARSE_FROM_ROOT_DEV_CONTAINER
```

Comment:
These top-level constants are a bit non standard, normally you'd put them inside a different global `let`.

Reply:
How would one go about testing different input variables (ie `directory`) without these lets?